modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/plugin/ArtifactPluginDescriptorLoader.java

```diff
@@ -6,48 +6,129 @@
  */
 package org.mule.runtime.module.deployment.internal.plugin;
 
-import static org.apache.commons.lang.StringUtils.removeEnd;
-import static org.mule.runtime.core.util.FileUtils.unzip;
-import static org.mule.runtime.api.util.Preconditions.checkArgument;
-
+import static java.lang.String.format;
+import static org.apache.commons.lang.StringUtils.isEmpty;
+import static org.mule.runtime.core.util.PropertiesUtils.loadProperties;
+import static org.mule.runtime.module.artifact.classloader.ArtifactClassLoaderFilterFactory.parseExportedResource;
+import static org.mule.runtime.module.artifact.classloader.DefaultArtifactClassLoaderFilter.EXPORTED_CLASS_PACKAGES_PROPERTY;
+import static org.mule.runtime.module.artifact.classloader.DefaultArtifactClassLoaderFilter.EXPORTED_RESOURCE_PROPERTY;
 import org.mule.runtime.deployment.model.api.plugin.ArtifactPluginDescriptor;
+import org.mule.runtime.module.artifact.descriptor.ArtifactDescriptorCreateException;
+import org.mule.runtime.module.artifact.descriptor.ClassLoaderModel.ClassLoaderModelBuilder;
 
 import java.io.File;
 import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
 
 /**
  * Loads a {@link ArtifactPluginDescriptor} from a file resource.
  *
  * @since 4.0
  */
-public class ArtifactPluginDescriptorLoader {
+abstract class ArtifactPluginDescriptorLoader {
```

Comment:
There must be an abstract or a parameterized test case to test subclasses of this class. (similar to what ArtifactPluginDescriptorLoaderTestCase used to do)